lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelection/BalanceSpec.hs

```diff
@@ -1095,6 +1062,163 @@ prop_performSelection mockConstraints (Blind params) coverage =
+            computeDeficitInOut params ===
+            computeDeficitInOut paramsTransformed
+
+        prop_selectionDeltaAllAssets =
+            counterexample "selectionDeltaAllAssets" $
+            selectionDeltaAllAssets result ===
+            selectionDeltaAllAssets resultTransformed
+
+        prop_selectionHasValidSurplus =
+            counterexample "selectionHasValidSurplus" $
+            selectionHasValidSurplus constraints result .&&.
+            selectionHasValidSurplus constraints resultTransformed
+
+    -- Checks that the transformation is correct.
+    prop_transformation =
+        counterexample "transformation correct" $
+        conjoin $ if null (view #outputsToCover params)
+        then
+            [ length (view #outputsToCover paramsTransformed) === 1
+            , length (view #outputsCovered resultTransformed) === 0
+            ]
+        else
+            -- If the initial list of outputs is non-empty, then no
+            -- transformation should take place:
+            [ params === (paramsTransformed & over #outputsToCover F.toList)
+            , resultTransformed === (result & over #outputsCovered F.toList)
+            ]
+
+    constraints :: SelectionConstraints
+    constraints = unMockSelectionConstraints mockConstraints
+
+    paramsTransformed :: SelectionParamsOf (NonEmpty TxOut)
+    paramsTransformed = view #paramsTransformed report
+
+    result :: SelectionResultOf (NonEmpty TxOut) TokenBundle
+    result = expectRight $ view #result report
+
+    resultTransformed :: SelectionResultOf [TxOut] TokenBundle
+    resultTransformed = expectRight $ view #resultTransformed report
+
+    -- Provides a report of how 'performSelectionEmpty' has transformed
+    -- both the parameters and result of 'mockPerformSelectionNonEmpty'.
+    --
+    report = performSelectionEmpty f constraints params
+      where
+        f constraints' params' = withTransformationReport params'
+            $ runIdentity
+            $ mockPerformSelectionNonEmpty constraints' params'
+
+-- | Provides a report of how function 'f' transforms function 'g', where
+--   function 'f' modifies both the parameters and the result of 'g'.
+--
+data TransformationReport paramsTransformed result resultTransformed =
+    TransformationReport
+        { paramsTransformed :: paramsTransformed
+            -- ^ The transformed parameters.
+        , result :: result
+            -- ^ The untransformed result.
+        , resultTransformed :: resultTransformed
+            -- ^ The transformed result.
+        }
+    deriving Generic
+
+-- | A functor instance on 'TransformationReport' that allows the final result
+--   value to be transformed by a function transformer.
+--
+instance Functor (TransformationReport paramsTransformed result) where
+    fmap f (TransformationReport paramsTransformed result resultUntransformed)
+        = TransformationReport paramsTransformed result (f resultUntransformed)
```

Comment:
**Suggestion: derive this instance automatically.**

This instance can actually be derived automatically, since the result to be transformed is the final type parameter.

Reply:
Fixed in faf943eda6b181917f0cce8ee3dcad440a290cb2.